backend/functions-disk.sh

```diff
@@ -346,6 +346,25 @@ delete_all_gpart()
 
 };
 
+# Function to check/delete a zpool as needed
+delete_old_zpool(){
+  local POOLNAME=$(grep -E "^zpoolName=" ${CFGF} | head -n 1 | cut -d "=" -f 2- | tr -d ' ' )
+  if [ -z "${POOLNAME}" ] ; then return ; fi
+  # Verify this pool isn't already in use
+  zpool list | grep -qw "${POOLNAME}"
+  if [ $? -eq 0 ] ; then
+    exit_err "The poolname: $POOLNAME is already in use locally!"
+  fi
+
+  # Verify this pool isn't lingering on another disk
+  zpool import | grep -qw "${POOLNAME}"
+  if [ $? -eq 0 ] ; then
+    # Need to clear out this pool name so we can re-create
+    rc_halt "zpool import -f -N ${POOLNAME}"
```

Comment:
My only concern is what if the pool is found, but can't be imported. Say its in a corrupt state, this will fail and break the installation.

Reply:
That is what already is happening with the current pc-sysinstall version. This just ensures that we run the zpool import/destroy routine before we start corrupting the pool.
I am not sure what the proper method for nuking a corrupt pool is aside from zero-ing out the disk itself. Seems to be a common problem/question about ZFS pool management.